core/dbt/helper_types.py

```diff
@@ -88,6 +90,63 @@ class NoValue(dbtClassMixin):
     novalue: NVEnum = field(default_factory=lambda: NVEnum.novalue)
 
 
+@dataclass
+class IncludeExclude(dbtClassMixin):
+    INCLUDE_ALL = ("all", "*")
+
+    include: Union[str, List[str]]
+    exclude: List[str] = field(default_factory=list)
+
+    def __post_init__(self):
+        if isinstance(self.include, str) and self.include not in self.INCLUDE_ALL:
+            raise ValidationException(
+                f"include must be one of {self.INCLUDE_ALL} or a list of strings"
+            )
+
+        if self.exclude and self.include not in self.INCLUDE_ALL:
+            raise ValidationException(
+                f"exclude can only be specified if include is one of {self.INCLUDE_ALL}"
+            )
+
+        if isinstance(self.include, list):
+            self._validate_items(self.include)
+
+        if isinstance(self.exclude, list):
+            self._validate_items(self.exclude)
+
+    def includes(self, item_name: str):
+        return (
+            item_name in self.include or self.include in self.INCLUDE_ALL
+        ) and item_name not in self.exclude
+
+    def _validate_items(self, items: List[str]):
+        pass
+
+
+class WarnErrorOptions(IncludeExclude):
+    # TODO: this method can be removed once the click CLI is in use
+    @classmethod
+    def from_yaml_string(cls, warn_error_options_str: Optional[str]):
+
+        # TODO: resolve circular import
+        from dbt.config.utils import parse_cli_yaml_string
```

Comment:
Getting a weird error from this method when I defined my config as:
```yml
# profiles.yml
config:
  warn_error_options:
    include:
      - NoNodesForSelectionCriteria
      - NothingToDo
```
```python
ipdb> warn_error_options_str
{'include': ['NoNodesForSelectionCriteria', 'NothingToDo']}
ipdb> from dbt.clients import yaml_helper
ipdb> yaml_helper.load_yaml_text(warn_error_options_str)
*** TypeError: a string or stream input is required
```
I could be doing something really dumb? I do think it's valid yaml, though, which is why I'm not seeing the nicer exception

Reply:
Good catch! Fixed 👍